src/libponyc/type/viewpoint.c

```diff
@@ -350,8 +350,14 @@ static void replace_type(ast_t** astp, ast_t* target, ast_t* with)
           switch(ast_id(eph))
           {
             case TK_EPHEMERAL:
-              a_with = consume_type(with, TK_NONE);
+            {
+              ast_t* c_with = consume_type(with, TK_NONE);
+              if (c_with != NULL)
+              {
+                a_with = with;
```

Comment:
is this correct, earlier a_with was already set to with.

should this be a_with = c_with?